main.go

```diff
@@ -140,16 +144,53 @@ func ansibleEnable() {
 	logrus.Infoln("Enabled Ansible-Puller")
 }
 
-func getAnsibleRepository(runDir string) error {
-	remoteURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), viper.GetString("http-url"))
+func getAnsibleRepository(context context.Context, runDir string) error {
+	httpURL := viper.GetString("http-url")
+	s3URI := viper.GetString("s3-uri")
```

Comment:
IMHO it should be uri or arn, not either. IIRC arn will give you the flexibility to work in china and us without any change besides the actual value

Reply:
I validated that S3 URIs should also give the flexibility to work in US/China thanks to the EC2 metadata integration in the v2 version of the golang sdk. In the past we've had to set the connection region but now we shouldn't have to.